doc/api_repos.rst

```diff
@@ -36,6 +28,15 @@ Repository Configuration
 
     Add a :class:`.Repo` to the repodict.
 
+  .. method:: add_new_repo(repoid, conf, baseurl=(), \*\*kwargs)
+
+    Initialize new :class:`.Repo` object and add it to the repodict. It requires ``repoid``
+    (string), and :class:`dnf.conf.Conf` object. Optionally it can be speciffied baseurl (list), and
+    additionally key/value pairs from `kwargs` to set additional attribute of the :class:`.Repo`
+    object. Variables in provided values (``baseurl`` or ``kwargs``) will be automatically
+    substituted using conf.substitutions (like ``$releasever``, ...). It returns the :class:`.Repo`
+    object.
```

Comment:
I know it's a copy paste, but this paragraph could use some English improvements :slightly_smiling_face: (let me know if I should attempt it)